### PR TITLE
Graph version upgrade

### LINKED ISSUE #8: [tech-debt] Upgrade subgraph graph-cli version
graph-cli version was downgraded from    
```
 "@graphprotocol/graph-cli": "0.25.0",
 "@graphprotocol/graph-ts": "0.24.1",
```
to:
```
 "@graphprotocol/graph-cli": "0.20.0",
 "@graphprotocol/graph-ts": "0.20.0",
```
because there were some breaking changes integrating univ3 prices

We need to restore latest cli version for being able to install matchstick-as for creating tests.

### PR DESCRIPTION
**Pull request by GabrielCamba**
We are paying the tech debt with this upgrading grahp-cli and graph-ts versions.
This new version will allow us to implement matchstick test framework and we will be able to deploy it on the decentralized network.

It's indexing here: https://thegraph.com/hosted-service/subgraph/gabrielcamba/something?selected=playground

I will remove the draft label once it finishes indexing.

closes #8

### CODE REVIEW

**Comment by GabrielCamba** [comment]
@anxolin @alfetopito : Do I have to take care about socket-security bot comments?

**Comment by anxolin** [comment]
> @anxolin @alfetopito : Do I have to take care about socket-security bot comments?

Keep the libs updated. If there's no security risk is fine

**Comment by GabrielCamba** [comment]
> Why is this PR still a draft? are u still having issues with this one?

Yes, I'm still facing issues. I'll remove the draft label once it has finished the deployment with no errors.

**Comment by alfetopito** [comment]
The code looks good, but the query results look weird

Seems like all prices are `0`. Or am I querying this wrong?
Screen Shot 2022-06-30 at 10 40 28

**Comment by anxolin** [comment]
Should this one go back to DRAFT?  also @GabrielCamba i think is good if you attach info in this PR given the last comment from Leandro. Is hard to follow on these topics otherwise

**Comment by GabrielCamba** [comment]
> The code looks good, but the query results look weird
> 
> Seems like all prices are `0`. Or am I querying this wrong? Screen Shot 2022-06-30 at 10 40 28
Deployed here: 
https://thegraph.com/hosted-service/subgraph/gabrielcamba/something?version=pending
The error should be fixed now. ;)

File: src/utils/staticTokenDefinition.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class StaticTokenDefinition {
       'DGD',
       BigInt.fromI32(9)
     )
-    staticDefinitions.push(tokenDGD)
+    staticDefinitions[0] = tokenDGD
```
Comment (alfetopito) [inline src/utils/staticTokenDefinition.ts:35]: Why this change?

File: src/scripts/deploy.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ series([
     confirm({
       message: chalk.red('Are you sure you want to deploy?'),
       callbackYes() {
-        exec(`graph deploy ${SUBGRAPH} --node https://api.thegraph.com/deploy/ --ipfs https://api.thegraph.com/ipfs/`, (error) => {
+        exec(`graph deploy ${SUBGRAPH} --product hosted-service`, (error) => {
```
[the remark this answers is not in the document]
Comment (GabrielCamba) [reply src/scripts/deploy.js:34]: yes, newer version  fails with same command, so I removed unnecessary params and it worked.

File: src/uniswapMappings/uniswapPoolFactory.ts [hunk 3]
```diff
@@ -42,7 +42,7 @@ export function handlePoolCreated(event: PoolCreated): void {
     token0.decimals = decimals.toI32()
     token0.priceEth = ZERO_BD
     token0.priceUsd = ZERO_BD
-    token0.allowedPools = []
+    token0.allowedPools = new Array<string>(0)
```
[the remark this answers is not in the document]
Comment (GabrielCamba) [reply src/uniswapMappings/uniswapPoolFactory.ts:45]: It's the same but I wanted to be explicit with the type to prevent failures.

File: src/utils/staticTokenDefinition.ts (hunk 1 above)
Comment (GabrielCamba) [reply src/utils/staticTokenDefinition.ts:35]: push was throwing an error, and since all of them are fixed and assigned in that file I went for that approach. Let me know if you find any reason and you want me to investigate further to keep using the push.

Comment (alfetopito) [reply src/utils/staticTokenDefinition.ts:35]: No need, was just curious
